versions.lock

```diff
@@ -20,7 +20,7 @@ com.squareup:javapoet:1.13.0 (2 constraints: 2b113eee)
 io.dropwizard.metrics:metrics-core:4.1.1 (2 constraints: c1130f66)
 io.dropwizard.metrics:metrics-jvm:4.1.1 (1 constraints: 08050336)
 net.bytebuddy:byte-buddy:1.12.17 (3 constraints: 5a1ccd22)
-org.checkerframework:checker-qual:3.25.0 (3 constraints: d524d761)
+org.checkerframework:checker-qual:3.26.0 (3 constraints: d524d761)
```

Comment:
Looks like checkerframework 3.26.0 updated return types to be more specific in https://github.com/typetools/checker-framework/pull/5296 (specifically https://github.com/typetools/checker-framework/blame/master/dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/LocalVariableNode.java#L63
 changed return type from `Element` to `VariableElement`) which is causing a `NoSuchMethodError` at https://github.com/uber/NullAway/blame/master/nullaway/src/main/java/com/uber/nullaway/dataflow/CoreNullnessStoreInitializer.java#L65 

NullAway currently depends on checkerframework 3.24.0, rebuilding against 3.26.0 may resolve this.

```
> Task :tritium-core:compileJava FAILED
compiler message file broken: key=compiler.misc.msg.bug arguments=11.0.16.1, {1}, {2}, {3}, {4}, {5}, {6}, {7}
com.google.common.util.concurrent.ExecutionError: java.lang.NoSuchMethodError: 'javax.lang.model.element.Element org.checkerframework.nullaway.dataflow.cfg.node.LocalVariableNode.getElement()'
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2053)
	at com.google.common.cache.LocalCache.get(LocalCache.java:3966)
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3989)
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4950)
	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:4956)
	at com.uber.nullaway.dataflow.DataFlow.dataflow(DataFlow.java:157)
	at com.uber.nullaway.dataflow.DataFlow.finalResult(DataFlow.java:235)
	at com.uber.nullaway.dataflow.AccessPathNullnessAnalysis.getNonnullFieldsOfReceiverAtExit(AccessPathNullnessAnalysis.java:155)
	at com.uber.nullaway.NullAway.guaranteedNonNullForConstructor(NullAway.java:1867)
	at com.uber.nullaway.NullAway.checkConstructorInitialization(NullAway.java:1844)
	at com.uber.nullaway.NullAway.checkFieldInitialization(NullAway.java:1697)
	at com.uber.nullaway.NullAway.matchClass(NullAway.java:1402)
	at com.google.errorprone.scanner.ErrorProneScanner.processMatchers(ErrorProneScanner.java:449)
	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:547)
	at com.google.errorprone.scanner.ErrorProneScanner.visitClass(ErrorProneScanner.java:150)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:808)
	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:74)
	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:48)
	at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
	at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:113)
	at jdk.compiler/com.sun.source.util.TreeScanner.visitCompilationUnit(TreeScanner.java:144)
	at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:560)
	at com.google.errorprone.scanner.ErrorProneScanner.visitCompilationUnit(ErrorProneScanner.java:150)
	at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:591)
	at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:56)
	at com.google.errorprone.scanner.Scanner.scan(Scanner.java:58)
	at com.google.errorprone.scanner.ErrorProneScannerTransformer.apply(ErrorProneScannerTransformer.java:43)
	at com.google.errorprone.ErrorProneAnalyzer.finished(ErrorProneAnalyzer.java:152)
	at jdk.compiler/com.sun.tools.javac.api.MultiTaskListener.finished(MultiTaskListener.java:132)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1414)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.flow(JavaCompiler.java:1361)
	at jdk.compiler/com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:960)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.lambda$doCall$0(JavacTaskImpl.java:104)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.handleExceptions(JavacTaskImpl.java:147)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:100)
	at jdk.compiler/com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:94)
	at org.gradle.internal.compiler.java.IncrementalCompileTask.call(IncrementalCompileTask.java:89)
	at org.gradle.api.internal.tasks.compile.AnnotationProcessingCompileTask.call(AnnotationProcessingCompileTask.java:94)
	at org.gradle.api.internal.tasks.compile.ResourceCleaningCompilationTask.call(ResourceCleaningCompilationTask.java:57)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:54)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:39)
	at org.gradle.api.internal.tasks.compile.daemon.AbstractDaemonCompiler$CompilerWorkAction.execute(AbstractDaemonCompiler.java:135)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:63)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:49)
	at org.gradle.workers.internal.AbstractClassLoaderWorker$1.create(AbstractClassLoaderWorker.java:43)
	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:100)
	at org.gradle.workers.internal.AbstractClassLoaderWorker.executeInClassLoader(AbstractClassLoaderWorker.java:43)
	at org.gradle.workers.internal.FlatClassLoaderWorker.run(FlatClassLoaderWorker.java:32)
	at org.gradle.workers.internal.FlatClassLoaderWorker.run(FlatClassLoaderWorker.java:22)
	at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:87)
	at org.gradle.workers.internal.WorkerDaemonServer.run(WorkerDaemonServer.java:56)
	at org.gradle.process.internal.worker.request.WorkerAction$1.call(WorkerAction.java:138)
	at org.gradle.process.internal.worker.child.WorkerLogEventListener.withWorkerLoggingProtocol(WorkerLogEventListener.java:41)
	at org.gradle.process.internal.worker.request.WorkerAction.run(WorkerAction.java:135)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:182)
	at org.gradle.internal.remote.internal.hub.MessageHubBackedObjectConnection$DispatchWrapper.dispatch(MessageHubBackedObjectConnection.java:164)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:414)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.NoSuchMethodError: 'javax.lang.model.element.Element org.checkerframework.nullaway.dataflow.cfg.node.LocalVariableNode.getElement()'
	at com.uber.nullaway.dataflow.CoreNullnessStoreInitializer.methodInitialStore(CoreNullnessStoreInitializer.java:65)
	at com.uber.nullaway.dataflow.CoreNullnessStoreInitializer.getInitialStore(CoreNullnessStoreInitializer.java:50)
	at com.uber.nullaway.dataflow.AccessPathNullnessPropagation.initialStore(AccessPathNullnessPropagation.java:203)
	at com.uber.nullaway.dataflow.AccessPathNullnessPropagation.initialStore(AccessPathNullnessPropagation.java:135)
	at org.checkerframework.nullaway.dataflow.analysis.ForwardAnalysisImpl.initInitialInputs(ForwardAnalysisImpl.java:342)
	at org.checkerframework.nullaway.dataflow.analysis.AbstractAnalysis.init(AbstractAnalysis.java:376)
	at org.checkerframework.nullaway.dataflow.analysis.ForwardAnalysisImpl.performAnalysis(ForwardAnalysisImpl.java:102)
	at com.uber.nullaway.dataflow.DataFlow$1.load(DataFlow.java:93)
	at com.uber.nullaway.dataflow.DataFlow$1.load(DataFlow.java:85)
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3533)
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2282)
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2159)
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2049)
	... 68 more
```

Reply:
Oof, I suspect NullAway, error-prone, and baseline-error-prone will need to upgrade in lock-step

Reply:
Yeah, I have a feeling those 3 will need to be tightly coupled on versions. I can pin checkerframework in tritium for now to unblock excavator on other deps. 

Filed https://github.com/uber/NullAway/issue/670 / https://github.com/uber/NullAway/pull/671 to bump checkerframework in NullAway.

Reply:
Actually, I think we can remove the checkerframework pin in tritium now, and we can just rely on versions propagating from gradle-baseline.